client/components/status/status.js

```diff
@@ -108,6 +183,23 @@ export default class Status extends React.Component {
             </div>
           </div>
         </div>
+        {captivePortalForm && (window.top === window.self) &&
+          <>
+            <form ref={this.formRef} method={captivePortalForm.method || "post"}
+              id="cp-form"
+              action={captivePortalForm.action || ""}
+              target="owisp-auth-iframe"
+              className="owisp-auth-hidden"
+            >
+              <input readOnly type="text" name={captivePortalForm.fields.username || ""} value={username} />
+              <input readOnly type="text" name={captivePortalForm.fields.password || ""} value={password} />
+              {captivePortalForm.additional_fields.length > 0 ? captivePortalForm.additional_fields.map(field =>
+                <input readOnly type="text" name={field.name} value={field.value} />
+              ) : null}
+            </form>
+            <iframe onLoad={this.handleIframe} ref={this.ifameRef} name="owisp-auth-iframe" className="owisp-auth-hidden" title="owisp-auth-iframe" />
+          </>
```

Comment:
also here you can help me out refreshing my react fu, why is this needed?

Reply:
ref is passed to get a reference to that element inside react class, I'm submitting the form in this iframe and have added this onLoad handler which checks for  reply param and 404 after the response is received from captive protal and loaded in the Iframe.

Reply:
also here can we add a comment?